spectrum-offchain-lm/src/executor.rs

```diff
@@ -160,12 +161,44 @@ where
                 let ctx = self.make_context(pool.box_id()).await;
                 info!("Running against {} with {}", pool, ctx);
                 let run_result = match (ord.clone(), bundles.first().cloned()) {
-                    (Order::Deposit(deposit), _) => deposit
-                        .try_run(pool.clone(), (), ctx)
-                        .map(|(tx, next_pool, bundle)| {
-                            (tx, next_pool, vec![bundle], None, OrderType::Deposit)
-                        })
-                        .map_err(|err| err.map(Order::Deposit)),
+                    (Order::Deposit(deposit), _) => {
+                        // Try to get token names from node
+                        let pool_id_bytes = <Vec<u8>>::from(TokenId::from(pool.1.pool_id));
+                        let pool_id_encoding = base16::encode_lower(&pool_id_bytes);
+                        let token_details = {
+                            let reward_token_name = self
+                                .network
+                                .get_token_minting_info(pool.1.budget_rem.token_id)
+                                .await;
+                            let liquidity_token_name =
+                                self.network.get_token_minting_info(deposit.1.lq.token_id).await;
+                            if let (Ok(Some(reward_name)), Ok(Some(lq_name))) =
+                                (reward_token_name, liquidity_token_name)
+                            {
+                                (format!("{}_{}_YF", reward_name.name, lq_name.name),
+                                 format!(
+                                     "The representation of your share in the {}/{} (pool id: {}) yield farming pool on the Spectrum Finance platform.",
+                                     reward_name.name,
+                                     lq_name.name,
+                                     pool_id_encoding,
+                                 )
+                                )
+                            } else {
+                                (String::from("Spectrum YF staking bundle"), 
```

Comment:
Looks like this logic should be placed in a separate layer (service).

Reply:
Done.